### ISSUE TITLE
Import exported phenotype assignments back into phenotyper

### ISSUE THREAD

**Issue by andrew-weisman** [ncats/multiplex-analysis-web-apps#15]
Higher priority than most Issues

**Comment by djsmith17** [comment]
This is almost completed. There are just small hiccups to iron out. As it comes to completion, I was to confirm its implementation. As its set up now, I assume the following:

- The exported file to be imported will have previously been saved in the Outputs folder and is imported in the Data Import/Export page
- The app will recognize the type of file to be loaded by the name and columns. The name of the file will always start with phenotype_summary and have the columns species_name_short, phenotype, species_name_long,  species_count, and species_percent
- The phenotype_summary can be loaded before a cells dataset is loaded, but the moment a full cell dataset is loaded, any phenotype summary information will be overwritten.
- Once a cells dataset is loaded, the phenotype_summary can be loaded again. 
- When the phenotype_summary is loaded, the phenotype method is set to 'Custom'. The phenotype_summary will act as a dictionary for the values listed in 'species_name_short' and 'phenotype'. It will then apply a phenotype in the cells dataframe that has a matching species_name_short. This does mean that its possible that some, if not all of the species_names short will not match, and importing the phenotype_summary is only as good as defining the no markers present condition. 
- All the downstream processing continue as normal with updates to the phenotype summary table, assigned phenotype table and the spatial plot

Please let me know if this makes sense, or if you want me to make an adjustment

**Comment by andrew-weisman** [comment]
> The exported file to be imported will have previously been saved in the Outputs folder and is imported in the Data Import/Export page

Yes, and for just those reasons, please make sure you search both the input and output folder for phenotype assignments to load back in.

> The app will recognize the type of file to be loaded by the name and columns. The name of the file will always start with phenotype_summary and have the columns species_name_short, phenotype, species_name_long, species_count, and species_percent

What do you mean by type of file? To be clear, this would only be for custom phenotyping, not for marker or species. (We will have to figure out saving the multiaxial gates at some point FYI but not now, but that might indeed have something different in the name or columns or something.) So whatever columns you need to regenerate the table that holds the custom phenotype assignments for each species. BTW, now might be the time to make changes to the table names written out to the page: "Phenotype Summary" --> "Phenotype Assignments" and "Assigned Phenotypes" --> "Phenotype Summary", what do you think? Likewise save the file with the prefix "phenotype_assignments" instead of "phenotype_summary".

> The phenotype_summary can be loaded before a cells dataset is loaded, but the moment a full cell dataset is loaded, any phenotype summary information will be overwritten.

Sure if you want to the first point, yes to the second point especially since you can just re-load the phenotype assignments file.

> Once a cells dataset is loaded, the phenotype_summary can be loaded again.

Yes.

> When the phenotype_summary is loaded, the phenotype method is set to 'Custom'. The phenotype_summary will act as a dictionary for the values listed in 'species_name_short' and 'phenotype'. It will then apply a phenotype in the cells dataframe that has a matching species_name_short. This does mean that its possible that some, if not all of the species_names short will not match, and importing the phenotype_summary is only as good as defining the no markers present condition.

I quite understand the last clause but otherwise, yes, exactly. And further, once a phenotype assignments file is loaded, the user will have the option to tune the actual phenotype assignments.

> All the downstream processing continue as normal with updates to the phenotype summary table, assigned phenotype table and the spatial plot

Yes. Loading the phenotype assignments will be like a bulk version of a single custom phenotype assignment and the entire app will respond as usual.
